### PR TITLE
docs: Canvas updates adding why and how describing data needs

### PR DESCRIPTION
**Pull request by srperf**
**What is this feature?**

This PR does the following updates to the canvas documentation:
- Add descriptions of why a user may want to use a canvas panel
- Adds descriptions on how and data needs for the canvas panel

**Why do we need this feature?**

This helps users when they look for information on the canvas visualization

**Who is this feature for?**

Any user learning or improving the use of canvas panels

**Which issue(s) does this PR fix?**:

None

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 1]
```diff
@@ -30,11 +30,26 @@ refs:
...
+They are extensible visualizations that allow you to add and arrange elements wherever you want within unstructured static and dynamic layouts.
+This lets you design custom visualizations and overlay data in ways that aren't possible with standard Grafana visualizations, all within the Grafana UI.
 
-> We would love your feedback on Canvas. Please check out the [open Github issues](https://github.com/grafana/grafana/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Aarea%2Fpanel%2Fcanvas) and [submit a new feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=type%2Ffeature-request,area%2Fpanel%2Fcanvas&title=Canvas:&projects=grafana-dataviz&template=1-feature_requests.md) as needed.
+{{< video-embed src="/static/img/docs/canvas-panel/canvas-beta-overview-9-2-0.mp4" max-width="750px" alt="Canvas beta overview" >}}
 
-{{< video-embed src="/static/img/docs/canvas-panel/canvas-beta-overview-9-2-0.mp4" max-width="750px" caption="Canvas beta overview" >}}
+If you've used popular UI and web design tools, then designing canvases will feel very familiar.
+With all of these dynamic elements, there are few limits to what a canvas can display.
```
Comment (adela-almasan) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:40]: I know what you mean with this but to me it feels like we're gonna start listing the limits.. 😅

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:40]:
```suggestion
With all of these dynamic elements, there's almost no limit to what a canvas can display.
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 2]
```diff
@@ -30,11 +30,26 @@ refs:
...
+
+{{< admonition type="note" >}}
+We'd love your feedback on the canvas visualization. Please check out the [open Github issues](https://github.com/grafana/grafana/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Aarea%2Fpanel%2Fcanvas) and [submit a new feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=type%2Ffeature-request,area%2Fpanel%2Fcanvas&title=Canvas:&projects=grafana-dataviz&template=1-feature_requests.md) as needed.
+{{< /admonition >}}
+
+## Supported data formats
+
+The canvas visualization is unique in that it doesn't have any specific data requirements. You can even start drawing and configuring visual elements without providing any data. However, any data you plan to consume should be accessible through supported Grafana data sources and structured in a way that ensures smooth integration with your custom elements.
```
Comment (adela-almasan) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:48]: _drawing_ ?

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:48]: Better?
```suggestion
The canvas visualization is unique in that it doesn't have any specific data requirements. You can even start adding and configuring visual elements without providing any data. However, any data you plan to consume should be accessible through supported Grafana data sources and structured in a way that ensures smooth integration with your custom elements.
```

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 3]
```diff
@@ -30,11 +30,26 @@ refs:
...
+
+If your canvas is going to update in real time, your data should support refreshes at your desired intervals without degrading the user experience.
+
+You can provide data to the multiple _elements_ and _connections_, like text, colors, and background pattern images, available in the canvas visualization. These have different data requirements defined in the [Elements](#elements) and [Connections](#connections) sections.
```
Comment (adela-almasan) [inline docs/sources/panels-visualizations/visualizations/canvas/index.md:52]: > like text, colors, and background pattern images
Looks like we're referring to them as elements and connections? They're options. I'm having a hard time understing this entire section 😅

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:52]: @adela-almasan does this make sense? @srperf - does this align with what you intended?
```suggestion
You can provide data to the multiple _elements_ and _connections_ available in the canvas visualization to set options like text, colors, and background pattern images. These have different data requirements defined in the [Elements](#elements) and [Connections](#connections) sections.
```

Comment (adela-almasan) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:52]: maybe something along these lines?

```suggestion
You can tie [Elements](#elements) and [Connections](#connections) to data through options like text, colors, and background pattern images, etc. available in the canvas visualization.
```

**Review by adela-almasan — APPROVED** [review]

**Pull request merged**